src/Microsoft.Health.Fhir.Core/Features/Search/Converters/IdentifierToTokenSearchValueConverter.cs

```diff
@@ -25,15 +25,28 @@ protected override IEnumerable<ISearchValue> Convert(ITypedElement value)
             string stringValue = value.Scalar("value") as string;
             string system = value.Scalar("system") as string;
             string type = value.Scalar("type.text") as string;
+            if (!string.IsNullOrEmpty(system) || !string.IsNullOrEmpty(type) || !string.IsNullOrEmpty(stringValue))
+            {
+                // Based on spec: http://hl7.org/fhir/search.html#token,
+                // the text for identifier is specified by Identifier.type.text.
+                yield return new TokenSearchValue(system, stringValue, type);
+            }
 
             if (string.IsNullOrEmpty(stringValue))
             {
                 yield break;
             }
```

Comment:
This should be the first check, we don't need to return anything if the value isn't there in either case

Reply:
Why?
Any reason why we don't want to have system only searches?

Let's say I have identifier:

        "identifier": [
          {
            "system":"cool-system"
          }
      ]
and I do search

?identifier = cool-system|

if I break on value been empty, where gonna be no record in system to search on.

Reply:
Sounds fair. In this case, do we need this clause at all? Can we have just `Identifier.system` and/or `Identifier.type.coding.system`?

Reply:
I don't want to create IdentifierOfTypeSearchValues, because it require presence of all three components. There can be multiple of SearchValues because I need to iterate over CodeableConcept.coding which is array, and it's easier to check value once than put it under foreach loop